dmapiclient/data.py

```diff
@@ -181,12 +181,12 @@ def set_framework_result(self, supplier_id, framework_slug, is_on_framework, use
             user=user,
         )
 
-    def register_framework_agreement_returned(self, supplier_id, framework_slug, user, agreement_details=None):
+    def register_framework_agreement_returned(self, supplier_id, framework_slug, user, uploader_user_id=None):
```

Comment:
We require older functionality for returning agreements for gcloud-7 to
still work (which does not require a uploader_user_id). The API only
requires the field to be set for frameworks with a framework agreement
version.

On 13 July 2016 at 15:23, Alexey Bezhan <notifications@github.com> wrote:

> In dmapiclient/data.py
> <https://github.com/alphagov/digitalmarketplace-apiclient/pull/28#discussion_r70634346>
> :
>
> > @@ -181,12 +181,12 @@ def set_framework_result(self, supplier_id, framework_slug, is_on_framework, use
> >              user=user,
> >          )
> >
> > -    def register_framework_agreement_returned(self, supplier_id, framework_slug, user, agreement_details=None):
> > +    def register_framework_agreement_returned(self, supplier_id, framework_slug, user, uploader_user_id=None):
>
> Should it be optional? As far as I understand, API requires the field to
> be set.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/alphagov/digitalmarketplace-apiclient/pull/28/files/4acc146c8558a525d643fde5ec6ef5289f1a955e#r70634346>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AG5MvVr5GmjJc_WqT8iTGN7bOLAP7bqDks5qVPTOgaJpZM4JJVCx>
> .
>